internal/api/dto/price.go

```diff
@@ -14,6 +14,7 @@ import (
 type CreatePriceRequest struct {
 	Amount             string                   `json:"amount"`
 	Currency           string                   `json:"currency" validate:"required,len=3"`
+	PriceUnit          string                   `json:"price_unit,omitempty"`
```

Comment:
if we are doing just this, then essentially the caller is responsible to make sure that they convert the `amount` field to the desired value?

Reply:
can follow same patter here as well maybe:
price_unit_code
price_unit_amount

but my only concern with that is that we need to then make amount and currency optional which is like making 99% case optional for 1% case.